backend/integration_tests/src/account_linking_tests.rs

```diff
@@ -0,0 +1,122 @@
+    );
+
+    let created_linking_code = match create_linking_code_response {
+        CreateAccountLinkingCodeResponse::Success(code) => code,
+        response => panic!("'create_account_linking_code' error: {response:?}"),
+    };
+
+    assert!(created_linking_code.value.len() == 6);
+    assert!(created_linking_code.is_valid(now_millis(env)));
+    assert_eq!(created_linking_code.user_id, user.user_id);
+
+    //
+    // ---- Fetch the linking code again to ensure it's the same ----
+    //
+    let get_linking_code_response = client::identity::get_account_linking_code(
+        env,
+        user_auth.auth_principal,
+        canister_ids.identity,
+        &identity_canister::get_account_linking_code::Args {},
+    );
+
+    let fetched_linking_code = match get_linking_code_response {
+        GetAccountLinkingCodeResponse::Success(code) => code,
+        response => panic!("'get_account_linking_code' error: {response:?}"),
+    };
+
+    assert_eq!(fetched_linking_code, created_linking_code);
+
+    //
+    // ---- Initialise new identity to link ----
+    //
+    let (new_principal, new_pub_key) = random_delegated_principal(canister_ids.sign_in_with_email);
+
+    //
+    // ---- Verify that the linking code can be used to link accounts ----
+    //
+
+    // In a real scenario, to avoid running an update call, we first verify
+    // that the linking code is valid with a query. Only after that we proceed
+    // to link the accounts using an update call.
+    let VerifyAccountLinkingCodeResponse(verify_success) = client::identity::verify_account_linking_code(
+        env,
+        new_principal,
+        canister_ids.identity,
+        &identity_canister::verify_account_linking_code::Args {
+            code: created_linking_code.value.clone(),
+        },
+    );
+
+    assert!(verify_success);
+
+    //
+    // ---- Link account auth methods using the code ----
+    //
+    let account_linking_response = client::identity::link_with_account_linking_code(
+        env,
+        new_principal,
+        canister_ids.identity,
+        &identity_canister::link_with_account_linking_code::Args {
+            code: created_linking_code.value.clone(),
+            public_key: new_pub_key,
+            webauthn_key: None,
+        },
+    );
+
+    match account_linking_response {
+        LinkWithAccountLinkingCodeResponse::Success => {}
+        LinkWithAccountLinkingCodeResponse::Error(err) => panic!("Account linking failed with: {err:#?}"),
+    }
+
+    //
+    // ---- Verify that the linking code is not returned anymore ----
+    //      (code has been removed from the map of active codes)
+    //
+    let get_linking_code_response = client::identity::get_account_linking_code(
+        env,
+        user_auth.auth_principal,
+        canister_ids.identity,
+        &identity_canister::get_account_linking_code::Args {},
+    );
+
+    assert_eq!(get_linking_code_response, GetAccountLinkingCodeResponse::NotFound);
+}
```

Comment:
Maybe add a final check that the account has been linked successfully by querying for the account using the new principal.